src/DocumentFormat.OpenXml/SimpleTypes/OpenXmlComparableReferenceType.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace DocumentFormat.OpenXml
+{
+    /// <summary>
+    /// Represents a comparable and equatable reference.
+    /// </summary>
+    /// <typeparam name="T">The type of the value.</typeparam>
+    public abstract class OpenXmlComparableReferenceType<T> : OpenXmlSimpleType,
```

Comment:
The name feels a little weird. Can we follow the previous pattern and go with `OpenXmlComparableSimpleType`?

Reply:
Yes, it is a little weird, so I am open for better names. But let me explain the thinking behind the current name.

On the `master` branch, we have `OpenXmlSimpleType` and `OpenXmlSimpleValue<T>` where the latter derives from the former. The subclasses representing value types (e.g., `DecimalValue`) and `EnumValue<T>` derive from `OpenXmlSimpleValue<T>`. The string-valued (reference) types (e.g., `StringValue`, `HexBinaryValue`) and `ListValue<T>` derive from `OpenXmlSimpleType`.

Now, to not introduce a breaking change, I needed to introduce:

1. one subclass of `OpenXmlSimpleValue<T>` (and therefore a grandchild of `OpenXmlSimpleType`) for all those comparable and equatable value types, hence the name `OpenXmlComparableValueType<T>`; and
2. one subclass of `OpenXmlSimpleType` that would become the superclass of the "reference types" `StringValue`, `HexBinaryValue`, and `Base64BinaryValue`.

For the second, the best name I could come up with was `OpenXmlComparableReferenceType`, because that was in contrast to `OpenXmlComparableValueType`, which was kind of given by its parent class `OpenXmlSimpleValue`.

Now, since both of those new, abstract base classes are "comparable simple types", we should not call just one of them `OpenXmlComparableSimpleType`.

Having said that, I am totally not wedded to the current names. If we find something that is logical, I am fine with that.

Reply:
Yeah, naming is hard 😆 I see your thought process and I think the issue I'm having with the name is "ValueType"/"ReferenceType" doesn't imply a kind of `OpenXmlSimpleType`. I want to make sure it's easy to see the relationship in the names (if possible). How about `OpenXmlComparableSimpleValue<T>` and `OpenXmlComparableSimpleReference<T>`. Not sure if they should be suffixed with "Type", but I think what I'm trying to get is a name that is obviously a subclass of `OpenXmlSimpleType`.

Reply:
`OpenXmlComparableSimpleValue<T>` and `OpenXmlComparableSimpleReference<T>` would be fine for me. I think I thought about the SimpleReference as well, but wasn't sure whether that introduces an unwanted connotation. But if you are fine with those, we can certainly also use those two.

And if you like `OpenXmlComparableSimpleType<T>` better, we could also go with that. As noted in my other comment, my only thought would be whether `OpenXmlComparableSimpleValue<T>` should derive from `OpenXmlComparableSimpleType<T>`, if possible.

Reply:
I like the first two. I agree they're not ideal, but it's a good compromise to keep naming consistent.

If we can get `OpenXmlComparableSimpleValue<T>` to derive from `OpenXmlComparableSimpleType<T>` that would be awesome, but I think it would break the hierarchy. Note the PR has a `package-compat-analyzer` check I wrote to validate that we don't break binary compatibility, so if you're not sure, you can submit an update and see what that analyzer says.